core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -270,6 +277,18 @@ static QueryStateMachine beginWithTicker(
             session = session.beginTransactionId(transactionId, transactionManager, accessControl);
         }
 
+        if (getRetryPolicy(session) == TASK) {
+            // encryption is mandatory for fault tolerant execution as it relies on an external storage to store intermediate data generated during an exchange
```

Comment:
I do not think we should enforce that. It should be configurable, but enabled by default. I can easily think of a deployment scenario when you do not care about encryption at all if storage is not really "external"